python_modules/dagster/dagster/_core/workspace/load_target.py

```diff
@@ -102,15 +105,15 @@ def get_origins_from_toml(
                 location_name=dagster_block.get("code_location_name"),
             ).create_origins()
         elif "modules" in dagster_block and is_valid_modules_list(dagster_block.get("modules")):
-            origins = []
+            origins: list[ManagedGrpcPythonEnvCodeLocationOrigin] = []
             for module in dagster_block.get("modules"):
                 if module.get("type") == "module":
                     origins.extend(
                         ModuleTarget(
                             module_name=module.get("name"),
                             attribute=None,
                             working_directory=os.getcwd(),
-                            location_name=dagster_block.get("code_location_name"),
+                            location_name=module.get("location_name"),
                         ).create_origins()
```

Comment:
Before, the location name for a module was loaded from `tool.dagster.code_location_name` directly (variable `dagster_block` refers to the `[tool.dagster]` table in pyproject.toml).
It would read this kind of config:
```toml
[tool.dagster]
code_location_name="foobar"
modules = [
    { type = "module", name = "module_a.definitions" },
    { type = "module", name = "module_b.definitions" },
]
```
This seems off. And when running it only the last module (in this example `module_b.definitions`) is loaded as code location to the dagster webserver. Therefore, I would consider `location_name=dagster_block.get("code_location_name")` to be a bug, which is addressed by this PR.